documentation/smg/16_serialiser.md

```diff
@@ -0,0 +1,229 @@
+# The serialiser
+
+In order to efficiently transfer MATLAB information between storage, be it via networked interaction (such as MPI) or to disc efficiently, the information is first "serialised" into a representation. This document details the operation of the serialisers defined in:
+```
+<HERBERT_ROOT>/herbert_core/utilities/misc/hlp_serialise
+<HERBERT_ROOT>/herbert_core/utilities/misc/hlp_deserialise
+<HERBERT_ROOT>/_LowLevelCode/cpp/cpp_communicator/serialiser/c_serialise
+<HERBERT_ROOT>/_LowLevelCode/cpp/cpp_communicator/serialiser/c_deserialise
+```
+
+The serialised format has recently been updated to be more consistent and to have greater capabilities such as serialising arrays of structs/objects.
+
+# New format
+
+## Standard header format
+
+The new style has a standardised header format depending on the shape of the data (with the exception of function handles).
+This format allows extension to support arrays of objects and structs in a way which the old serialiser couldn't.
+It also greatly simplifies the tag structure and allows it to align with the C++ MEX API data types.
+Data in these cases is serialised in the linearised order as defined by `data(:)` and the array reshaped by the deserialiser.
```

Comment:
Sorry, do I know what 'data(:)' is?

Reply:
`array(:)` is standard MATLAB notation for treating a multidimensional-array as a vector (i.e. in memory order). 
e.g.
```
a = eye(4); %2-D identity
a(:)
```

Reply:
@oerc0122 Sorry, that wasn't my point. 'data' appears to be a particular variable, which I don't think has been defined. I just added the (:) to quote the complete string. Is 'data' supposed to be the Data following the header? I think this is the first time it has been identified in this document as a MATLAB variable as opposed to a general entity.

Reply:
Sorry, I'll try to make that more clear